rust/src/ldap/ldap.rs

```diff
@@ -158,24 +170,43 @@ impl LdapState {
         })
     }
 
-    fn parse_request(&mut self, input: &[u8]) -> AppLayerResult {
+    fn parse_request(&mut self, flow: *const Flow, stream_slice: StreamSlice) -> AppLayerResult {
+        let input = stream_slice.as_slice();
         if input.is_empty() {
             return AppLayerResult::ok();
         }
 
+        let _pdu = Frame::new(
+            flow,
+            &stream_slice,
+            input,
+            input.len() as i64,
+            LdapFrameType::Pdu as u8,
+            None,
+        );
+        SCLogDebug!("ts: pdu {:?}", _pdu);
+
         let mut start = input;
         while !start.is_empty() {
             match ldap_parse_msg(start) {
                 Ok((rem, msg)) => {
-                    start = rem;
                     let mut tx = self.new_tx();
+                    let tx_id = tx.id();
                     let request = LdapMessage::from(msg);
                     tx.complete = match request.protocol_op {
                         ProtocolOp::UnbindRequest => true,
                         _ => false,
                     };
                     tx.request = Some(request);
                     self.transactions.push_back(tx);
+
+                    let consumed = start.len() - rem.len();
+                    start = rem;
+                    if let Some(frame) = &self.request_frame {
```

Comment:
This creates only one frame in the case there are 2 PDUs in a single TCP packet, does it not ?

Reply:
Yes it does.

Reply:
It seems to me that `Frame::new` is outside the loop